File: android/app/src/main/java/com/customrnviewdemo/reactpackage/ViewPackage.java
package com.customrnviewdemo.reactpackage;

import com.customrnviewdemo.view.CustomAndroidTextViewManager;
import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * Created by wangjiajie on 2017/9/15.
 */

public class ViewPackage implements ReactPackage {
    @Override
    public List<NativeModule> createNativeModules(ReactApplicationContext reactContext) {
        return Collections.emptyList();
    }

    @Override
    public List<ViewManager> createViewManagers(ReactApplicationContext reactContext) {
        return Arrays.<ViewManager>asList(new CustomAndroidTextViewManager());
    }
}

File: android/app/src/main/java/com/customrnviewdemo/reactpackage/ViewPackage.kt
package com.customrnviewdemo.reactpackage

import com.customrnviewdemo.manager.CircleManager
import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager

import java.util.Arrays

/**
 * Created by wangjiajie on 2017/9/16.
 */

class ViewPackage : ReactPackage {
    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> =
            emptyList()

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> = Arrays.asList<ViewManager<*, *>>(
            CircleManager()
    )
}
